articles/prettier-setup/index.md

```diff
@@ -0,0 +1,188 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/prettier-setup/
+title: How to setup code formatting for Node.js apps using Prettier and ESLint
+description: How to setup code formatting for Node.js apps using Prettier and ESLint
+author: aman-saxena
+date: 2020-10-19T00:00:00-05:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/prettier-setup/hero.jpg
+    alt: image example
+
+---
+In open-source development, handling a large number of files across every development stage is quite a tedious task. These stages consist of enormous chunks of code that not only require a great amount of time needed for formatting but also differ in style with different individuals in the team.
+<!--more-->
+### Setting up code formatting
+Have you ever wondered how to cut short formatting time with just one-click? Any alternatives to prevent this monotonous task?
+
+Getting rid of the hassles of writing clean code and maintaining consistency across a project on specific code style guidelines, **Prettier** is one of the most prominent tools out there in which you can bank upon! With Prettier and githook integration, you the code is formatted within a single-click. Prettier, along with the support of **ESLint**, is a deadly combination that increases productivity manifolds.
+
+ This guide includes all the extensive information to get used to the code formatting style in the open source circuit. Let's get started by answering in-detail all the why(s), what(s) & how(s)!
+
+### Table of Contents
+- What is Prettier?
+- Why is Prettier so useful?
+- How does it work?
+- How to install Prettier?
+- How to configure Prettier to work with ESLint?
+- How to format all your code with just one-click?
+- Further Reading
+
+### A Complete Roadmap
+
+### What is Prettier?
+
+Prettier is an opinionated code formatter with support for JavaScript, GraphQL, Angular and other popular Community Plugins. It is highly established in the open-source ecosystem with more than 2 million dependent repositories on GitHub. You can measure its efficiency by the fact that it has more than 6000 dependent packages on npm. Many of the technology giants like Facebook, React, Dropbox and PayPal, etc. use it quite extensively. It also has support for various IDE, viz. VS Code, Web Storm, Atom and Vim.
+
+![gif-1](./implement.gif)
+[Prettier Magic](https://glebbahmutov.com/blog/images/prettier/projectA.gif)
+
+### Why is Prettier so useful?
+
+- **Building and enforcing a style guide**: In open source development, having common style guidelines is a highly desirable feature in a project, but achieving that is a little tricky process. Prettier not only provides a perfect platform to build the style guide from scratch with suitable requirements but also helps in maintaining it throughout the life cycle of the project.
```

Comment:
This sentence that explains prettier could be useful in the start of the article.